src/models/__tests__/profile.test.ts

```diff
@@ -184,147 +185,151 @@ describe("Profile codec", () => {
 
-    expect(isLeft(result)).toBeTruthy();
-    if (isLeft(result)) {
-      expect(result.value).toEqual("error");
+    expect(E.isLeft(result)).toBeTruthy();
+    if (E.isLeft(result)) {
+      expect(result.left.kind).toEqual("COSMOS_ERROR_RESPONSE");
     }
   });
 });
 
-describe("update", () => {
+describe("updateProfile", () => {
   it("should update an existing profile", async () => {
-    const clientMock: any = {
-      createDocument: jest.fn((_, newDocument, __, cb) => {
-        cb(undefined, {
-          ...newDocument,
-          _self: "self",
-          _ts: 123
-        });
-      }),
-      readDocument: jest.fn((_, __, cb) => cb(undefined, aRetrievedProfile))
+    const containerMock = ({
+      items: {
+        create: jest.fn().mockReturnValue(
+          Promise.resolve({
+            resource: {
+              ...aRetrievedProfile,
+              id: `${aFiscalCode}-${"0".repeat(15)}1`,
+              version: 1,
+              email: "new@example.com"
+            }
+          })
+        )
+      }
+    } as unknown) as Container;
+
+    const model = new ProfileModel(containerMock);
+
+    const aProfileWithDifferentEmail = {
+      ...aRetrievedProfile,
+      email: "new@example.com" as EmailString
     };
 
-    const model = new ProfileModel(clientMock, profilesCollectionUrl);
+    const result = await model.update(aProfileWithDifferentEmail)();
 
-    const result = await model.update(
-      aRetrievedProfile.fiscalCode,
-      aRetrievedProfile.fiscalCode,
-      p => {
-        return {
-          ...p,
-          email: "new@example.com" as EmailString
-        };
-      }
+    expect(containerMock.items.create).toHaveBeenCalledTimes(1);
+    expect(containerMock.items.create).toHaveBeenCalledWith(
+      expect.objectContaining({
+        kind: "INewProfile",
+        fiscalCode: aProfileWithDifferentEmail.fiscalCode,
+        email: aProfileWithDifferentEmail.email
+      }),
+      expect.anything()
     );
 
-    expect(clientMock.createDocument).toHaveBeenCalledTimes(1);
-    expect(clientMock.createDocument.mock.calls[0][1].kind).toBeUndefined();
-    expect(clientMock.createDocument.mock.calls[0][2]).toHaveProperty(
-      "partitionKey",
-      aFiscalCode
-    );
-    expect(isRight(result)).toBeTruthy();
-    if (isRight(result)) {
-      expect(result.value.isSome()).toBeTruthy();
-      if (isSome(result.value)) {
-        const updatedProfile = result.value.value;
-        expect(updatedProfile.fiscalCode).toEqual(aRetrievedProfile.fiscalCode);
-        expect(updatedProfile.id).toEqual(`${aFiscalCode}-${"0".repeat(15)}1`);
-        expect(updatedProfile.version).toEqual(1);
-        expect(updatedProfile.email).toEqual("new@example.com");
-      }
+    expect(E.isRight(result)).toBeTruthy();
+    if (E.isRight(result)) {
+      const updatedProfile = result.right;
+      expect(updatedProfile.fiscalCode).toEqual(aRetrievedProfile.fiscalCode);
```

Comment:
ditto. if we want to test values because they are post processed by some code we have to extract this values and avoid repeating them.